v2/yarpcprotobuf/codec.go

```diff
@@ -54,6 +56,41 @@ func (c *protoCodec) Encode(res interface{}) (*yarpc.Buffer, error) {
 	return nil, yarpcerror.InternalErrorf("tried to encode a non-proto.Message in protobuf codec")
 }
 
+func (c *protoCodec) EncodeError(err error) (*yarpc.Buffer, error) {
+	if res == nil {
+		return &yarpc.Buffer{}, nil
+	}
+
+	info := yarpcerror.ExtractInfo(err)
+	if details == nil {
+		return nil, nil
+	}
+	p = spb.Status{
+		Code: info.Code,
+		Message: info.Message,
+	}
+	if details := yarpcerror.ExtractDetails(err); details != nil {
+		if m, ok := details.(proto.Message); ok {
```

Comment:
optional nit: we can switch on types, not sure if this would clean this up or not, but it would remove the ok checks